src/__test__/server.test.js

```diff
@@ -0,0 +1,196 @@
+  beforeAll(startServer);
+  afterAll(stopServer);
+  afterEach(() => User.remove({}));
+
+  describe('POST /api/users', () => {
+    test('POST - It should respond with a 200 status ', () => {
+      const userToPost = {
+        name: faker.name.findName(),
+        location: faker.address.city(),
+        bicepsRPM: faker.random.number(300),
+        tricepsRPM: faker.random.number(300),
+      };
+      return superagent.post(apiURL)
+        .send(userToPost)
+        .then((response) => {
+          expect(response.status).toEqual(200);
+          expect(response.body.name).toEqual(userToPost.name);
+          expect(response.body.location).toEqual(userToPost.location);
+          expect(response.body.bicepsRPM).toEqual(userToPost.bicepsRPM);
+          expect(response.body.tricepsRPM).toEqual(userToPost.tricepsRPM);
+          expect(response.body._id).toBeTruthy();
+        });
+    });
+    test('POST - It should respond with a 400 status ', () => {
+      const UserToPost = {
+        location: faker.address.city(),
+        bicepsRPM: faker.random.number(300),
+        tricepsRPM: faker.random.number(300),
+      };
+      return superagent.post(apiURL)
+        .send(UserToPost)
+        .then(Promise.reject)
+        .catch((response) => {
+          expect(response.status).toEqual(400);
+        });
+    });
+    test('POST - 409 due to duplicate name', () => {
+      return createUserMock()
+        .then((savedUser) => {
+          const mockCategory = {
+            name: savedUser.name,
+            location: savedUser.location,
+          };
+          return superagent.post(apiURL)
+            .send(mockCategory);
+        })
+        .then(Promise.reject)
+        .catch((error) => {
+          expect(error.status).toEqual(409);
+        });
+    });
+  });
+
+  describe('GET /api/users/:id', () => {
+    test('GET - should respond with 200 if there are no errors', () => {
+      let userToTest = null;
+      return createUserMock()
+        .then((user) => {
+          userToTest = user;
+          return superagent.get(`${apiURL}/${user._id}`);
+        })
+        .then((response) => {
+          expect(response.status).toEqual(200);
+          expect(response.body.name).toEqual(userToTest.name);
+          expect(response.body.location).toEqual(userToTest.location);
+          expect(response.body.bicepsRPM).toEqual(userToTest.bicepsRPM);
+          expect(response.body.tricepsRPM).toEqual(userToTest.tricepsRPM);
+        });
+    });
+    test('GET - should respond with 404 if there is no user to be found', () => {
+      return superagent.get(`${apiURL}/THisIsAnInvalidId`)
+        .then(Promise.reject)
+        .catch((response) => {
+          expect(response.status).toEqual(404);
+        });
+    });
+    test('GET - should respond with an array if no id specified.', () => {
+      return superagent.get(apiURL)
+        .then((response) => {
+          expect(Array.isArray(response.body)).toBeTruthy();
```

Comment:
You could just write this like so since `Array.isArray` returns true or false if the input is an array.

```
expect(Array.isArray(response.body)).toEqual(true)
```